src/extensions/exception-autocapture/index.ts

```diff
@@ -117,7 +118,13 @@ export class ExceptionObserver {
         this.startIfEnabled()
     }
 
-    captureException(errorProperties: Properties) {
+    captureException(error: Error, additionalProperties: Properties = {}) {
+        const errorEventArgs: ErrorEventArgs = [error.message, undefined, undefined, undefined, error]
+        const errorProperties = errorToProperties(errorEventArgs)
+        this._captureException({ ...errorProperties, ...additionalProperties })
+    }
```

Comment:
yep, I think this is the 3% bump in bundle... we're pulling in errorToProperties which would otherwise not be imported unless exception capture is configured

I _think_ this should check if error stuff is on the window and log if not
and we give people a local config to turn it on (if it's not already possible)

if lots of people use it we can make it the default or have remote config to allow manual but not auto capture???

or we add this and stand the 3% bump for everyone but that feels wrong IMHO open to challenge

Reply:
Certainly wasn't my intention to increase the default bundle size... I actually thought I'd avoid it by moving `errorToProperties` into this file rather than `src/posthog-core.ts`.

I pushed another commit that hopefully moves things in the right direction but not sure how to fully evaluate this. Is there any prior art for how we do this with something similar?

Do you think there should be a local config to enable manual capture if someone doesn't have autocapture enabled? It feels like a bit of a gotcha to me but get that it might be necessary if we want to ensure the bundle size stays small for folks not using it. I wonder what the flow looks like for manually showing surveys, we should make sure that both function the same way at the very least

Reply:
the core capture in surveys is part of the main SDK - this is always available and anyone can use it. The extension is all the code for showing surveys.

What imo would be ideal here (and what we did for surveys as well) is to figure out how the manual exception capture API would look like, and re-use that inside the extension as well (instead of the manual `_captureException` function) - I think the interface you have already looks reasonable, and the personURL can be just another part of `additionalProperties`.

The contention here is that the `errorToProperties` function (+dependents) is pretty big, while the surveys core code is pretty small, so taking the same approach might not make the most sense.

As such, I think there are ~two~ three options:

1. Keep basic manual exception capture as part of main SDK (which does not do stack trace parsing). And when they enable autocapture/stack trace parsing - we do this parsing to get all the details of the error.
2. Since autocapture is 10% of bundle and stack trace parsing is 3%, we can make two extensions: One for manual capture, one for autocapture. Enabling autocapture also enables manual capture.
3. Make stack trace parsing part of the core SDK, 3% isn't that big of a hit (what you already currently have).


I'm currently leaning towards (1), specially because least effort currently, and gives us enough information (by users having to turn on autocapture to get the stack trace details) to make a decision whether (2) or (3) is worth it. Also doesn't affect current bundle size for existing users 👌 . My hypothesis is that we'll find out people who want to do manual capture would be very happy with having autocapture as well, so would anyways turn it on.

Reply:
re-reading, looks like Paul said the exact same thing but in fewer words 🙈 🙈

Reply:
it is a rare event that I'm the most concise person 🤣